programs/08-wasm-eth/src/custom_query.rs

```diff
@@ -0,0 +1,79 @@
+use alloy_primitives::B256;
+use cosmwasm_std::{Binary, CustomQuery, Deps, QueryRequest};
+use ethereum_light_client::types::bls::{BlsPublicKey, BlsSignature, BlsVerify};
+use ethereum_utils::{ensure::ensure, hex::to_hex};
+use thiserror::Error;
+
+#[derive(serde::Serialize, serde::Deserialize, Clone)]
+#[serde(rename_all = "snake_case")]
+pub enum EthereumCustomQuery {
```

Comment:
What is this?

Reply:
This is for the custom query to bls verification (BlsVerify)

Reply:
Would love that. I don't want us to write this ourselves, but created an issue for it: https://github.com/cosmos/solidity-ibc-eureka/issues/163

If Union didn't want to do it rewrite it themselves, I doubt we should :D 

There is a rust library we use for tests (same that Union uses for their tests): https://github.com/Snowfork/milagro_bls, but it is not production ready (big warning on the repo about that).